pkg/resources/provisioning.cattle.io/v1/cluster/validator_test.go

```diff
@@ -200,3 +202,587 @@ func TestValidateMachinePoolName(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateSystemAgentDataDirectory(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name          string
+		cluster       *v1.Cluster
+		oldCluster    *v1.Cluster
+		shouldSucceed bool
+	}{
+		{
```

Comment:
Some test cases that would be good to add:
- old env var is set, new DataDirectories.System agent is set, new env var is still provided
- old env var is set, new DataDirectories.System agent is set but doesn't agree with the old value